src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -285,81 +298,246 @@ public override async Task<bool> PackageVersionExistsAsync(PackageURL purl, bool
+            return versionsList;
+        }
+    
+        /// <summary>
+        /// Parses the maven repository HTML to provide either a (version, published timestamp) mapping or a (file name, published timestamp) mapping
+        /// (depending on the provided baseUrl). 
+        /// </summary>
+        /// <returns></returns>
+        private async Task<Dictionary<string, DateTime?>>? DirectoryIndexStrategyAsync(string baseUrl, PackageURL purl, bool useCache)
+        {
+            HttpClient httpClient = CreateHttpClient();
+            var publishedTimestampDict = new Dictionary<string, DateTime?>();
+
+            try
+            {
+                string? content = await GetHttpStringCache(httpClient, baseUrl, useCache);
+
+                if (string.IsNullOrWhiteSpace(content))
+                {
+                    return null;
+                }
+
+                HtmlParser parser = new();
+                AngleSharp.Html.Dom.IHtmlDocument document = await parser.ParseDocumentAsync(content);
+
+                // Break the content down into its individual lines. Includes the parent directory and xml + hash files.
+                var htmlRowArray = document.QuerySelector("#contents").TextContent.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+
+                if (htmlRowArray == null)
+                {
+                    return null;
+                }
+
+                foreach (var row in htmlRowArray)
+                {
+                    // Split the content into its individual parts.
+                    // [0] - The version or file name (depending on the provided base URL)
+                    // [1] - The date it was published
+                    // [2] - The time it was published
+                    // [3] - The file size in bytes
+                    string[] rowParts = row.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+
+                    // Do not include the .xml and parent directory.
+                    if (!rowParts[0].Equals("../") && !rowParts[0].EndsWith(".xml") && !rowParts[0].EndsWith(".xml.md5") && !rowParts[0].EndsWith(".xml.sha1"))
+                    {
+                        // Try to parse the published timestamp.
+                        DateTime? publishedTimestamp = null;
+                        if (DateTime.TryParse($"{rowParts[1]} {rowParts[2]}", out DateTime publishDateTime))
+                        {
+                            publishedTimestamp = publishDateTime;
+                        }
+
+                        // Trim any '/' from the version or file name
+                        publishedTimestampDict.Add(rowParts[0].Replace("/", string.Empty), publishedTimestamp);
+                    }
+                }
+            }
+            catch (Exception e)
             {
-                return publishDateTime;
+                Logger.Trace($"Directory index strategy for {purl} was unsuccessful: {e.Message}");
```

Comment:
Whenever we log exceptions, it's best practice to provide the exception object directly to the logger: https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-exceptions

This allows the logger to report the full stack trace in a well-formatted way.

Reply:
Same note applies for all of the exception handlers that log exceptions

Reply:
Added!